### PR TITLE
Task-44573: Add the maximum number of characters counter for task description

### PR DESCRIPTION
**Pull request by wassimmhamdi**
Problem: The task description field has no indication on the description's maximum number of characters
Fix: we add the character counter to make a limitation of the task's description which is 2000 characters like the one of project description field

### CODE REVIEW

**Review by rdenarie — CHANGES_REQUESTED** [review]
Can you update the PR to resolv the conflict ?

**Review by rdenarie — CHANGES_REQUESTED** [review]
There is a lot of duplicated line in less file. Can you clean it ?

File: task-management/src/main/webapp/skin/css/tasks.less [hunk 1]
```diff
@@ -2723,7 +2797,7 @@
     }
 
     .activityCharsCount.tooManyChars {
-      display: none;
+      //display: none;
```
Comment (rdenarie) [inline task-management/src/main/webapp/skin/css/tasks.less]: Why this modification is needed ?

File: task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue [hunk 2]
```diff
@@ -47,10 +62,23 @@ export default {
   data() {
     return {
       inputVal: this.value,
-      editorReady: false
+      editorReady: false,
+      editorReady1: false,
```
Comment (rdenarie) [inline task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue]: Why do we use 2 variable with quite same name ?

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: task-management/src/main/webapp/skin/css/tasks.less (hunk 1 above)
Comment (wassimmhamdi) [reply task-management/src/main/webapp/skin/css/tasks.less]: to make sure that the characters counter is always appeared

File: task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue [hunk 3]
```diff
@@ -12,13 +12,28 @@
...
-      id="descriptionContent"
-      ref="editor"
-      v-model="inputVal"
-      class="d-none"
-      cols="30"
-      rows="10"></textarea>
+    <div
+      :class="charsCount > 0 ? '' : 'emptyEditor'"
+      class="projectDescription"
+      style="position: relative">
```
Comment (rdenarie) [inline task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue]: do we need this div ? is ist possible to put it on div .taskDescription ? 
In addition, the class of this div still mention project

**Review by rdenarie — CHANGES_REQUESTED** [review]
As we have already a component which do the same thing (ExoTaskEditor.vue), I wish that you try to use it.

I'm pretty sure than we can have the same mecanism of "clickToEdit" with using ExoTaskEditor.
Can you try this solution please ?

File: task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue (hunk 3 above)
Comment (wassimmhamdi) [reply task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue]: I tried to use  "exo-task-editor"  as a reusable component that do the same as  (ExoTaskEditor.vue),and it works correctly 
@rdenarie

File: task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue [hunk 4]
```diff
@@ -46,13 +48,27 @@ export default {
...
   computed: {
+    charsCount() {
+      return this.inputVal && this.$utils.htmlToText(this.inputVal).length || 0;
+    },
+    validLength() {
+      return this.charsCount <= this.maxLength;
+    },
+    buttonId() {
+      return `btn_${this.ckEditorType}`;
+    },
```
Comment (rdenarie) [inline task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue]: should be removed
```suggestion
```

**Review by rdenarie — CHANGES_REQUESTED** [review]
it seems that you have a duplication in less file. Can you check

File: task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue (hunk 4 above)
Comment (wassimmhamdi) [reply task-management/src/main/webapp/vue-app/taskDrawer/components/TaskDrawerComponents/TaskDescriptionEditor.vue]: I did it

File: task-management/src/main/webapp/skin/css/tasks.less [hunk 5]
```diff
@@ -2695,6 +2727,31 @@
...
+        }
+      }
+    }
+
+    .tooManyChars {
+      color: #bc4343!important;
+    }
+    .tooManyChars .uiIconMessageLength {
+      color: #bc4343!important;
+    }
```
Comment (rdenarie) [inline task-management/src/main/webapp/skin/css/tasks.less:2708]: This seems duplicated. Can you check ?

**Review by rdenarie — CHANGES_REQUESTED** [review]

File: task-management/src/main/webapp/skin/css/tasks.less (hunk 5 above)
Comment (wassimmhamdi) [reply task-management/src/main/webapp/skin/css/tasks.less:2708]: I resolved the problem

File: task-management/src/main/webapp/skin/css/tasks.less [hunk 6]
```diff
@@ -245,6 +245,7 @@
         .cke_bottom .cke_button_icon {
           filter: opacity(0.4);
         }
+
```
Comment (rdenarie) [inline task-management/src/main/webapp/skin/css/tasks.less]:
```suggestion
```

**Review by rdenarie — CHANGES_REQUESTED** [review]

**Review by rdenarie — APPROVED** [review]

**Pull request merged**